test/html.ts

```diff
@@ -51,6 +67,17 @@ describe('render', () => {
       render(main(fragment), surface)
       expect(surface.innerHTML).to.equal('<span>Hello Universe!</span>')
     })
+
+    it('renders DocumentFragments nested in sub templates nested in arrays', () => {
+      const sub = () => {
+        const frag = document.createDocumentFragment()
+        frag.appendChild(document.createElement('div'))
+        return html`<span>${frag}</span>`
+      }
+      const main = () => html`<div>${[sub(), sub()]}</div>`
+      render(main(), surface)
+      expect(surface.innerHTML).to.contain('<div><span><div></div></span><span><div></div></span></div>')
+    })
```

Comment:
This test demonstrates the bug. It's "fixed" by adding the `return` but because it is such a roundabout test I felt adding the other test would clarify matters.